### DISCUSSION TITLE
Super confusing GitHub Container Registry (need pullable public image)

### DISCUSSION THREAD

**Discussion by nazar-pc** [community/community#26861] [Packages]
<p>So my goal is to have a public image pullable without authentication/authorization.</p>
<p>The first (successfull) attempt with <code>docker/build-push-action@v1</code>, but it wasn’t pullable.</p>
<p>Now with combo of <code>docker/login-action@v1</code>, <code>docker/build-push-action@v2[with][load] = true</code> and just <code>docker push</code> I have pushed a few images to <code>ghcr.io</code> (under user that has admin access to organization), but still can’t pull them without authentication.</p>
<p>Here are those that should be pullable:</p>
<ul>
<li><code>ghcr.io/subspace/subspace-core-rust/subspace:updates</code></li>
<li><code>ghcr.io/subspace/subspace:updates</code></li>
</ul>
<p>Those were pushed a few minutes ago, but I can’t find them anywhere in UI and can’t pull anonymously.</p>
<p>Here is what my action file looks like:</p>
<pre><code class="lang-auto">    steps:
      - uses: actions/checkout@v2
      - name: Inject slug/short variables
        uses: rlespinasse/github-slug-action@v2.x
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
      - name: Cache Docker layers
        uses: actions/cache@v2
        with:
          path: /tmp/.buildx-cache
          key: ${{ runner.os }}-buildx-${{ github.sha }}
          restore-keys: |
            ${{ runner.os }}-buildx-${{ env.GITHUB_REF_SLUG }}
            ${{ runner.os }}-buildx-
      - uses: docker/login-action@v1
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ secrets.CR_PAT }}
      - name: Build container image
        uses: docker/build-push-action@v2
        with:
          load: true
          tags: |
            ghcr.io/subspace/subspace:${{ env.GITHUB_REF_SLUG }}
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache
      - name: Push container image
        run: |
          docker push ghcr.io/subspace/subspace:${{ env.GITHUB_REF_SLUG }}
</code></pre>

**Comment by airtower-luna** [comment]
<p>Images on GHCR are private by default, after creating a new image (not tag!) you need to set it to public before it can be pulled without authentication. <a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/configuring-access-control-and-visibility-for-container-images#configuring-visibility-of-container-images-for-an-organization">Configuring visibility of container images for an organization</a> should describe what you need. :slightly_smiling_face:</p>

**Comment by nazar-pc** [comment]
<p>I wish docs matched my UI. There is no “Package Settings” button in my UI. There is only “Edit package” dropdown that doesn’t contain anything like that.<br>
Moreover, I can’t find images pushed to <code>ghcr.io</code> anywhere in my UI, they are not in the list of packages.</p>
<p>This is why it is super confusing.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That’s odd, the equivalent for personal accounts worked perfectly for me on the weekend.</p>
<aside class="quote no-group" data-username="nazar-pc" data-post="1" data-topic="133334">
<div class="title">
<div class="quote-controls"></div>
 nazar-pc:</div>
<blockquote>
<p>I have pushed a few images to <code>ghcr.io</code> (under user that has admin access to organization)</p>
</blockquote>
</aside>
<p>Is that user who has admin access someone other than you, and they just provided the PAT? In that case you may need to ask them to make the package public.</p>

**Comment by nazar-pc** [comment]
<p>OK, that helped, thanks!<br>
But why is package that is inside of organization is only visible to a single user and not other admins unless you change access rules?<br>
Doesn’t make any sense to me.</p>
<p>Also I can now see “Package Settings” on these new images, but not on old that is on <code>docker.pkg.github.com</code>.</p>

**Comment by clarkbw** [comment]
<aside class="quote no-group" data-username="nazar-pc" data-post="5" data-topic="133334">
<div class="title">
<div class="quote-controls"></div>
 nazar-pc:</div>
<blockquote>
<p>But why is package that is inside of organization is only visible to a single user and not other admins unless you change access rules?<br>
Doesn’t make any sense to me.</p>
</blockquote>
</aside>
<p>We’ll be improving the defaults on this so you don’t see it as much without trying but we wanted to enable systems to have packages that aren’t visible to owners by default.  Many larger organizations have owners who are the IT managers for the system and not necessarily the ones who should have access to all resources. And people are using containers (generic storage) for things like PII (Personally Identifiable Information) with controlled access.  Owners will still be able to directly access the container but that will be logged in the audit log.</p>
<aside class="quote no-group" data-username="nazar-pc" data-post="5" data-topic="133334">
<div class="title">
<div class="quote-controls"></div>
 nazar-pc:</div>
<blockquote>
<p>Also I can now see “Package Settings” on these new images, but not on old that is on <code>docker.pkg.github.com</code> .</p>
</blockquote>
</aside>
<p>Yes, these settings will only be available to GHCR. The <code>docker.pkg.github.com</code> service is deprecated and won’t see these changes, it also inherits all permissions from the repository so these settings weren’t necessary.</p>

**Comment by nazar-pc** [comment]
<p>Looks like confusion is by design, this is even more sad.</p>
<p>I was able to get rid of the <code>docker.pkg.github.com</code> only after contacting support, which is suboptimal.</p>

**Comment by clarkbw** [comment]
<aside class="quote no-group" data-username="nazar-pc" data-post="7" data-topic="133334">
<div class="title">
<div class="quote-controls"></div>
 nazar-pc:</div>
<blockquote>
<p>I was able to get rid of the <code>docker.pkg.github.com</code> only after contacting support, which is suboptimal.</p>
</blockquote>
</aside>
<p>We’ll be adding support for deleting those images soon so you won’t have to contact support.</p>
